pkg/util/strategicpatch/patch.go

```diff
@@ -963,13 +972,170 @@ func HasConflicts(left, right interface{}) (bool, error) {
 	}
 }
 
+// MergingMapsHaveConflicts returns true if the left and right JSON interface
```

Comment:
The implementation started out that way, but it quickly became unmanageable. Also, the existing method is being used in its current form, so going that route meant create a stub that offered the old API over the new one, passing in a nil structure for type information, and then plumbing that through the implementation. In the end, it was simpler and cleaner to leave the existing method unchanged, and to write the new one straightforwardly using strategic merge patch semantics.